src/handlers/feature_handlers.cpp

```diff
@@ -4054,6 +4055,92 @@ void ltLensFlareHandler::Install() {
     }
 }
 
+/*
+    ltLightHandler
+*/
+
+gfxTexture* RedGlowTexture;
+gfxTexture* HeadlightTexture;
+gfxTexture* SirenTexture;
+gfxTexture* AmbientTexture;
+
+ltLight* ltLightHandler::Constructor()
+{
+    auto light = reinterpret_cast<ltLight*>(this);
+    light->Default();
+
+    if (!ltLight::GlowTexture)
+        ltLight::GlowTexture = gfxGetTexture("lt_glow", true);
+
+    light->HighlightTexture = gfxGetTexture("lt_highlight", true);
+
+    if (!RedGlowTexture)
+        RedGlowTexture = gfxGetTexture("s_red_glow", true);
+
+    if (!HeadlightTexture)
+        HeadlightTexture = gfxGetTexture("lt_glow_headlight", true);
+
+    if (!SirenTexture)
+        SirenTexture = gfxGetTexture("lt_glow_siren", true);
+
+    if (!AmbientTexture)
+        AmbientTexture = gfxGetTexture("lt_glow_ambient", true);
+
+    return light;
+}
+
+void ltLightHandler::ShutdownLights()
+{
+    if (ltLight::GlowTexture)
+    {
+        gfxFreeTexture((gfxTexture*)ltLight::GlowTexture);
```

Comment:
Unncessary cast?

Reply:
It must be there otherwise the game will crash on freeing it so the cast is necessary.